client/components/datapicker/datapicker.mjs

```diff
@@ -0,0 +1,168 @@
+const date = new Date();
+const calendarBox = document.querySelector(".dropdown");
+const dropdown = document.getElementById("myDropdown");
+const monthDays = document.querySelector(".days");
+const nonActiveInput = document.querySelector(".calendar-input-label")
+
+
+function renderCalendar() {
+  date.setDate(1);
+
+  const lastDay = new Date(
+    date.getFullYear(),
+    date.getMonth() + 1,
+    0
+  ).getDate();
+
+  const prevLastDay = new Date(
+    date.getFullYear(),
+    date.getMonth(),
+    0
+  ).getDate();
+
+  const firstDayIndex = date.getDay();
+  const lastDayIndex = new Date(
+    date.getFullYear(),
+    date.getMonth() + 1,
+    0
+  ).getDay();
+
+  const nextDays = 7 - lastDayIndex - 1;
+
+  const months = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ];
+
+  const year = date.getFullYear();
+
+  document.querySelector(".date h1").innerHTML =
```

Comment:
be aware that querySelector, selects a group of matching selectors. So it will update innerHTML of any `.date h1` on page, even if it doesn't belong to datepicker. 

To avoid this, best practice would be to use id.
Or at least make your class more specific: prefix it with `.datepicker-`

Reply:
changed on "datepicker-date h1"